bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/TypeChecker.java

```diff
@@ -2988,6 +3017,11 @@ private static boolean isEqual(Object lhsValue, Object rhsValue, List<ValuePair>
             case TypeTags.TABLE_TAG:
                 return rhsValTypeTag == TypeTags.TABLE_TAG &&
                         isEqual((TableValueImpl) lhsValue, (TableValueImpl) rhsValue, checkedValues);
+            case TypeTags.TYPE_REFERENCED_TYPE_TAG:
+                return checkValueEquals(lhsValue, rhsValue, checkedValues,
+                        ((BTypeReferenceType) lhsValType).getReferredType(), rhsValType);
+            default:
+                return false;
         }
         return false;
```

Comment:
Shall we move the `TypeTags.SERVICE_TAG` case on top of the `default` case and omit the return statement here?